linode/regions/framework_datasource_schema.go

```diff
@@ -0,0 +1,68 @@
+package regions
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/terraform-provider-linode/linode/helper/frameworkfilter"
+)
+
+var filterConfig = frameworkfilter.Config{
+	"capabilities": {APIFilterable: false},
+	"country":      {APIFilterable: false},
+	"status":       {APIFilterable: false},
+}
+
+var frameworkDataSourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description: "The data source's unique ID.",
+			Computed:    true,
+		},
+	},
+	Blocks: map[string]schema.Block{
```

Comment:
We may want to switch to non-block normal list style as discussed in the last Friday.

Reply:
Good point! I think the main advantage to this approach is that descriptions, etc. can be embedded in nested structures, which can come in handy for [automatic code completion](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform).

That said, I think consistency is key so I'm definitely open to any feedback on this approach. Does anyone else have any thoughts?

Reply:
Do we need to change the implementation of the filter schema, if we switch to a non-block style?

Reply:
@yec-akamai Assuming I'm understanding the question correctly,  I believe the implementation would change a good bit if we switched to non-block style schema for nested objects. Fortunately it shouldn't affect any existing data sources 👍

Reply:
So if we want to stick to non-block, we will switch to use a similar style as we do for the nested list currently, right? personally I think having the nested object here to take the advantage of descriptions and more details is better. Probably limit the usage for filter needed schemas only, will be helpful for consistency?

Reply:
> So if we want to stick to non-block, we will switch to use a similar style as we do for the nested list currently, right?
Yep, the alternative to the current block layout would be using the nested list structure that we use in our already migrated data sources.

> personally I think having the nested object here to take the advantage of descriptions and more details is better.
After giving it some thought I agree with this. Being able to include descriptions, etc. in nested fields can definitely come in handy with these data sources.

> Probably limit the usage for filter needed schemas only, will be helpful for consistency?
I think there's room for discussion on this but generally I think scoping this to filter schemas should be fine 👍

Reply:
> So if we want to stick to non-block, we will switch to use a similar style as we do for the nested list currently, right?

Yep, the alternative to the current block layout would be using the nested list structure that we use in our already migrated data sources.

> personally I think having the nested object here to take the advantage of descriptions and more details is better.

After giving it some thought I agree with this. Being able to include descriptions, etc. in nested fields can definitely come in handy with these data sources.

> Probably limit the usage for filter needed schemas only, will be helpful for consistency?

I think there's room for discussion on this but generally I think scoping this to filter schemas should be fine 👍